src/Aspire.Hosting/Eventing/DistributedApplicationEventing.cs

```diff
@@ -0,0 +1,41 @@
+namespace Aspire.Hosting.Eventing;
+
+internal class DistributedApplicationEventing : IDistributedApplicationEventing
+{
+    private readonly Dictionary<Type, List<DistributedApplicationEventSubscription>> _subscriptions = new();
+
+    public async Task PublishAsync<T>(T @event, CancellationToken cancellationToken) where T : IDistributedApplicationEvent
+    {
+        if (_subscriptions.TryGetValue(typeof(T), out var subscriptions))
+        {
+            foreach (var subscription in subscriptions)
+            {
+                await subscription.Callback(@event, cancellationToken).ConfigureAwait(false);
```

Comment:
Do we want these subscriptions to run in serial order like this? Each one needing to wait for the previous to finish.

Reply:
It is a good question. I think whether it runs serially or not is something that the publisher of the event gets to decide and not something that we enforce in eventing itself.